src/IW/Sync/Update.hs

```diff
@@ -14,27 +14,36 @@ import IW.App (WithError)
 import IW.Core.Repo (Repo (..))
 import IW.Db (WithDb, upsertRepos, updateRepoCategories)
 import IW.Effects.Cabal (MonadCabal (..), getCabalCategories)
-import IW.Sync.Search (fetchAllHaskellRepos, fromGitHubRepo)
+import IW.Sync.Search (searchAllHaskellRepos)
 
 
--- | This function fetches the latest repos from the GitHub API, parses their @.cabal@ files,
--- and upserts them into the database.
-fetchAndUpsertRepos
+-- | This function fetches repos from the GitHub API within a specified date range,
+-- parses their @.cabal@ files, and upserts them into the database.
+syncRepos
     :: forall env m.
        ( MonadCabal m
        , MonadUnliftIO m
        , WithDb env m
        , WithLog env m
        , WithError m
        )
-    => m ()
-fetchAndUpsertRepos = do
-    gitHubRepos <- fetchAllHaskellRepos
-    let repos = map fromGitHubRepo gitHubRepos
+    => Integer -- ^ Starting interval
+    -> m ()
+syncRepos interval = do
+    repos <- searchAllHaskellRepos interval
     upsertRepos repos
-    mapConcurrently_ fetchAndUpdateCategories repos
-  where
-    fetchAndUpdateCategories :: Repo -> m ()
-    fetchAndUpdateCategories Repo{..} = do
-        categories <- getCabalCategories repoOwner repoName
-        updateRepoCategories repoOwner repoName categories
+    mapConcurrently_ syncCategories repos
```

Comment:
`syncRepos` now look simple. But I guess the main complexity is hidden inside `searchAllHaskellRepos`.

Reply:
@chshersh yeah I thought this was better. I felt like all that functionality I had in syncRepos belonged to the actual search function instead.